Tests/NIOHTTP2Tests/TestUtilities.swift

```diff
@@ -62,6 +62,32 @@ extension XCTestCase {
         } while operated
     }
 
+    /// Have two `NIOAsyncTestingChannel` objects send and receive data from each other until
+    /// they make no forward progress.
+    func interactInMemory(_ first: NIOAsyncTestingChannel, _ second: NIOAsyncTestingChannel, file: StaticString = #filePath, line: UInt = #line) async throws {
+        var operated: Bool
+
+        func readBytesFromChannel(_ channel: NIOAsyncTestingChannel) async -> ByteBuffer? {
+            guard let data = try? await assertNoThrowWithValue(await channel.readOutbound(as: ByteBuffer.self)) else {
+                return nil
+            }
+            return data
+        }
+
+        repeat {
+            operated = false
+
+            if let data = await readBytesFromChannel(first) {
+                operated = true
+                try await assertNoThrow(try await second.writeInbound(data), file: (file), line: line)
+            }
+            if let data = await readBytesFromChannel(second) {
+                operated = true
+                try await assertNoThrow(try await first.writeInbound(data), file: (file), line: line)
```

Comment:
The parentheses around `file` were to squash a warning a long time ago, they're not necessary anymore.

```suggestion
                try await assertNoThrow(try await second.writeInbound(data), file: file, line: line)
            }
            if let data = await readBytesFromChannel(second) {
                operated = true
                try await assertNoThrow(try await first.writeInbound(data), file: file, line: line)
```